Prettify output of REST calls in DocService if JSON

hueneburg opened the pull request:
Adds a new button in the doc service to allow pretty print of the response.

<img src="https://user-images.githubusercontent.com/7841099/77813713-0f555780-70ee-11ea-902a-793d34450914.png" width="300" /><img src="https://user-images.githubusercontent.com/7841099/77813719-1e3c0a00-70ee-11ea-91c4-75a291f9fb17.png" width="300" />

I've created a test endpoint that creates an object with 100 elements and each element is an object with three values:
<img src="https://user-images.githubusercontent.com/7841099/77813896-ac64c000-70ef-11ea-8740-5a132e96b530.png" width="300" /><img src="https://user-images.githubusercontent.com/7841099/77813903-b7b7eb80-70ef-11ea-8d01-739a8227bc59.png" width="300" />

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/line/armeria?pullRequest=2635) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/line/armeria?pullRequest=2635) before we can accept your contribution.<br/><hr/>**Armin Huneburg** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/line/armeria?pullRequest=2635) it.</sub>

anuraaga commented:
Thanks! Let's use `prettify` like we do elsewhere which doesn't have trouble round tripping certain corner cases of JSON

https://github.com/line/armeria/blob/84362c4ae1b85f11e4b0fd422c6db9d25e513484/docs-client/src/lib/transports/grpc-unframed.ts#L57

Also, we are applying it automatically for gRPC but instead of having a button how about applying it automatically for any response with JSON content type?

hueneburg commented:
Changed commit author from work email to private one...

hueneburg commented:
How do I retest, because locally that step succeeds...

hueneburg commented:
@anuraaga I didn't add it automatically because the fetching of the response is done differently and only returns `String`.

trustin commented:
> I didn't add it automatically because the fetching of the response is done differently and only returns String.

Could you elaborate on this a little bit?

hueneburg commented:
Ok, since you've mentioned you do it automatically for grpc, I now noticed that there is a transport class for http, which I didn't notice before. I've added an automatic prettifier if the content type contains `json`. Should I leave the prettify button in case the content type does not contain json but a json is returned?

trustin commented:
Great! Yes, I think we can keep the prettify button if not prettified already.

minwoox commented:
> I think we can keep the prettify button if not prettified already.

Shouldn't we just remove the button because otherwise, it will cause an error when the response is not JSON?

ikhoon commented:
I think we can improve two points:
* If `prettify` could not handle the input string, it would be better to show the original text than blank.
* `prettify` maybe not parse JSON array. 
![Kapture 2020-03-31 at 14 59 25](https://user-images.githubusercontent.com/1866157/77992244-a6c1e100-7360-11ea-8d3e-78e78bdbca46.gif)

anuraaga commented:
I'd lean towards removing the button too at least for now - it seems so rare to not have the correct content-type that it would probably mostly be just noise on the UI.

trustin commented:
How about trying to prettify the content regardless of the content type? If prettifying fails (because it's not a JSON), then we can just give up. Then, the prettify button can be gone.

hueneburg commented:
I've updated it to return the original string if the prettified version is empty and I removed the prettify button.

@trustin you suggested to always prettify the response but it is hard to assess whether the entire response is returned or not if we get a non-json response...

trustin commented:
Thanks a lot for your patience, @hueneburg. Great job!
